underscore.js

```diff
@@ -779,13 +779,30 @@
 
   // Memoize an expensive function by storing its results.
   _.memoize = function(func, hasher) {
+    function getCache() {
+      if (Object.create) {
+        return Object.create(null);
+      }
+      return {};
+    }
+
+    function hasKey(cache, address) {
+      // Objects created with `Object.create(null)` don't have a constructor
+      // property.
+      if (cache.constructor === Object) {
+        return _.has(cache, address);
+      } else {
+        return (typeof cache[address] !== 'undefined');
```

Comment:
Makes sense!
Should I add a test for this case?
Right now everything is passing.